common/ledger/blockledger/fileledger/factory_test.go

```diff
@@ -87,3 +101,38 @@ func TestMultiReinitialization(t *testing.T) {
 	require.Equal(t, 2, len(f.ChannelIDs()))
 	f.Close()
 }
+
+func TestRemove(t *testing.T) {
+	mockBlockStore := &mock.BlockStoreProvider{}
+	dir, err := ioutil.TempDir("", "fileledger")
+	require.NoError(t, err, "Error creating temp dir: %s", err)
+	defer os.RemoveAll(dir)
+
+	fileRepo, err := filerepo.New(dir, "remove")
+	require.NoError(t, err, "Error creating temp file repo: %s", err)
+	f := &fileLedgerFactory{
+		blkstorageProvider: mockBlockStore,
+		ledgers:            map[string]blockledger.ReadWriter{},
+		removeFileRepo:     fileRepo,
+	}
+
+	t.Run("success", func(t *testing.T) {
+		mockBlockStore.DropReturns(nil)
+		err = f.Remove("foo")
+		require.NoError(t, err, "Error removing channel")
+		require.Equal(t, 1, mockBlockStore.DropCallCount(), "Expected 1 Drop() calls")
+	})
+
+	t.Run("drop fails", func(t *testing.T) {
+		mockBlockStore.DropReturns(errors.New("oogie"))
+		err = f.Remove("foo")
+		require.EqualError(t, err, "oogie")
+	})
```

Comment:
If drop fails, the `foo.remove` remains. you have to check for that.

Reply:
🧹